### PR TITLE
Ngrx generator

### PR DESCRIPTION
**Pull request by JonnyBGod**
#### What type of pull request are you creating?
- [ ] Bug Fix
- [x] Enhancement
- [ ] Documentation

#### How many unit test did you write for this pull request?

2

Write a reason if none (e.g just fixed a typo):

#### Please add a description for your pull request:

Generate first NGRX implementation

#### Progress:

- [x] Actions
- [x] Effects
- [x] Reducers
- [x] Guards
- [x] Resolvers
- [x] Tests

supersedes https://github.com/mean-expert-official/loopback-sdk-builder/pull/374#event-1004905458

### CODE REVIEW

File: tests/ng2web/src/app/shared/sdk/resolvers/auth-account.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/* tslint:disable */
+import { Observable } from 'rxjs/Observable';
+import { Injectable } from '@angular/core';
+import { Resolve, ActivatedRouteSnapshot } from '@angular/router';
+import { Store } from '@ngrx/store';
+
+import { User } from '../models/user';
```
Comment (brannon-darby) [inline tests/ng2web/src/app/shared/sdk/resolvers/auth-account.ts]: do we want to use `User` or `Account`?

Comment (JonnyBGod) [inline tests/ng2web/src/app/shared/sdk/resolvers/auth-account.ts]: This is generated depending on the api configuration.

The test api is using Account.

File: lib/angular2/shared/resolvers/auth-account.ejs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/* tslint:disable */
+import { Observable } from 'rxjs/Observable';
+import { Injectable } from '@angular/core';
+import { Resolve, ActivatedRouteSnapshot } from '@angular/router';
+import { Store } from '@ngrx/store';
+
+import { User } from '../models/user';
```
Comment (brannon-darby) [inline lib/angular2/shared/resolvers/auth-account.ejs]: should User be dynamic?

File: lib/angular2/shared/reducers/auth.ejs [hunk 3]
```diff
@@ -0,0 +1,85 @@
+/* tslint:disable */
+import { Observable } from 'rxjs/Observable';
+import { Action } from '@ngrx/store';
+import { SDKToken } from '../models/BaseModels';
+import { User } from '../models/user';
```
Comment (brannon-darby) [inline lib/angular2/shared/reducers/auth.ejs]: User = <%- userModelName %>

**Comment by RicardoVaranda** [comment]
Thanks for the brilliant work guys, any ETA on this PR?

**Comment by JonnyBGod** [comment]
close in favor of (https://github.com/mean-expert-official/loopback-sdk-builder/pull/489)

**Pull request closed without merging**